rotkehlchen/exchanges/bitfinex.py

```diff
@@ -0,0 +1,982 @@
+                    case=case_,
+                )
+
+            try:
+                response_list = rlk_jsonloads_list(response.text)
+            except JSONDecodeError:
+                msg = f'{self.name} {case} returned invalid JSON response: {response.text}.'
+                log.error(msg)
+                self.msg_aggregator.add_error(
+                    f'Got remote error while querying {self.name} {case}: {msg}',
+                )
+                return []  # type: ignore # bug list nothing
+
+            results_ = self._deserialize_api_query_paginated_results(
+                case=case_,
+                options=call_options,
+                currency_map=currency_map,
+                raw_results=response_list,
+                processed_result_ids=processed_result_ids,
+            )
+            results.extend(cast(Iterable, results_))
+            # NB: Copying the set before updating it prevents losing the call args values
+            processed_result_ids = processed_result_ids.copy()
+            processed_result_ids.update({int(result.link) for result in results_})
+
+            if len(response_list) < limit:
+                break
+            # Update pagination params per endpoint
+            # NB: Copying the dict before updating it prevents losing the call args values
+            call_options = call_options.copy()
+            call_options.update({
+                'start': results[-1].timestamp * 1000,
+            })
+
+        return results
+
+    def _check_for_system_clock_not_synced_error(
+            self,
+            error_code: Optional[int] = None,
+    ) -> None:
+        if error_code == API_SYSTEM_CLOCK_NOT_SYNCED_ERROR_CODE:
+            raise SystemClockNotSyncedError(
+                current_time=str(datetime.now()),
+                remote_server=f'{self.name}',
+            )
+
+    @overload  # noqa: F811
+    def _deserialize_api_query_paginated_results(  # pylint: disable=no-self-use
+            self,
+            case: Literal['trades'],
+            options: Dict[str, Any],
+            currency_map: Dict[str, str],
+            raw_results: List[List[Any]],
+            processed_result_ids: Set[int],
+    ) -> List[Trade]:
+        ...
+
+    @overload  # noqa: F811
+    def _deserialize_api_query_paginated_results(  # pylint: disable=no-self-use
+            self,
+            case: Literal['asset_movements'],
+            options: Dict[str, Any],
+            currency_map: Dict[str, str],
+            raw_results: List[List[Any]],
+            processed_result_ids: Set[int],
+    ) -> List[AssetMovement]:
+        ...
+
+    def _deserialize_api_query_paginated_results(
+            self,
+            case: Literal['trades', 'asset_movements'],
+            options: Dict[str, Any],
+            currency_map: Dict[str, str],
+            raw_results: List[List[Any]],
+            processed_result_ids: Set[int],
+    ) -> Union[List[Trade], List[AssetMovement]]:
+        """TODO
```

Comment:
You forgot a TODO here. You don't need huge docstrings. Even something small is fine. Also lack of docstring for a self-explanatory function is also fine. Just not a "TODO" docstring.